### PR TITLE
cmt: init at 1.17

### PR DESCRIPTION
**Pull request by sjfloat**
###### Motivation for this change

Introduction of new package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` (New pkg -- no dependents)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sjfloat** [comment]
Does anyone know why I can't request review on this? It's greyed out.

File: pkgs/applications/audio/cmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    make targets
+  '';
+
+  installPhase = ''
+    substituteInPlace Makefile --replace /usr/lib $out/lib
+    mkdir -p $out/lib/ladspa
+    make install
+  '';
+
+  buildInputs = [ ladspaH ];
```
Comment (jonringer) [inline pkgs/applications/audio/cmt/default.nix]: generally inputs are listed before phases, and you should prefer to use the pre/post hooks over writing new phases:

```suggestion
  buildInputs = [ ladspaH ];

  preBuild = ''
    cd src
  '';

  installFlags = [ "INSTALL_PLUGINS_DIR=${placeholder "out"}/lib/ladspa" ];
  preInstall = ''
    mkdir -p $out/lib/ladspa
  '';
```

**Review by jonringer — COMMENTED** [review]
is there supposed to be headers? all that's exported is a .so
```
$ tree ./results/cmt
./results/cmt
└── lib
    └── ladspa
        └── cmt.so
```

**Comment by jonringer** [comment]
> @jonringer Does anyone know why I can't request review on this? It's greyed out.

no idea, but I've noticed github having issues for a while

**Comment by sjfloat** [comment]
> > @jonringer Does anyone know why I can't request review on this? It's greyed out.
> 
> no idea, but I've noticed github having issues for a while

Thanks again!

**Comment by sjfloat** [comment]
> is there supposed to be headers? all that's exported is a .so
> 
> ```
> $ tree ./results/cmt
> ./results/cmt
> └── lib
>     └── ladspa
>         └── cmt.so
> ```

Excellent question! But no, these are plugins. Nothing is built against them.

**Review by jonringer — COMMENTED** [review]
do you mind squashing the changes commit into the original commit. Otherwise LGTM

**Comment by sjfloat** [comment]
> do you mind squashing the changes commit into the original commit. Otherwise LGTM

Not at all!

**Review by jonringer — APPROVED** [review]
LGTM

```
[1 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/82683
1 package built:
cmt
```
```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-82683-1]$ nix path-info -Sh ./results/cmt
/nix/store/kdvk7lwcdgcj2k03djjh1w7im5694768-cmt	  37.5M
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build cmt

**Pull request merged**